edx_event_bus_kafka/internal/producer.py

```diff
@@ -34,16 +36,20 @@
 # CloudEvent standard name for the event type header, see
 # https://github.com/cloudevents/spec/blob/v1.0.1/kafka-protocol-binding.md#325-example
 EVENT_TYPE_HEADER_KEY = "ce_type"
-ERROR_STRING_TEMPLATE = "Error publishing event to event bus. error={error} {context}"
 
-# constants for error messages
-FULL_TOPIC_DICT_KEY = "full_topic"
-EVENT_TYPE_DICT_KEY = "event_type"
-EVENT_KEY_DICT_KEY = "event_key"
-SIGNAL_DICT_KEY = "signal"
-INITIAL_TOPIC_DICT_KEY = "initial_topic"
-EVENT_KEY_FIELD_DICT_KEY = "event_key_field"
-EVENT_DATA_DICT_KEY = "event_data"
+
+def record_error(error, context):
+    """
+    Record an error in producing and event to both the monitoring system and the regular logs
+    """
+    try:
+        raise Exception(error)
+    except BaseException:
+        record_exception()
+        error_template = "Error publishing event to event bus. error={error} {context!r}"
+        logger.exception(
+            error_template.format(error=error, context=context)  # pylint: disable=logging-format-interpolation
+        )
```

Comment:
I think this could be simplified to

```suggestion
        logger.exception(f"Error publishing event to event bus. {error=} {context!r}")
```

Reply:
Not if I want to keep the pylint disable. it ends up going over the line limit

Reply:
Oh wait I misread. I'll try that.